homeassistant/components/roon/media_player.py

```diff
@@ -29,6 +29,7 @@
 from homeassistant.util.dt import utcnow
 
 from .const import DOMAIN
+from .event import RoonEventEntity
```

Comment:
Add the event entity in the event platform.

Reply:
This isn't straightforward.

The `media_players` in roon aren't static - the roon system creates them dynamically as devices become available (and deletes then too). So the media player creating code tracks what players come back from the roon api as things updates - and creates any that don't yet exist via a callback. You can see the code here:

```python
    @callback
    def async_update_media_player(player_data):
        """Add or update Roon MediaPlayer."""
        dev_id = player_data["dev_id"]
        if dev_id not in media_players:
            # new player!
            media_player = RoonDevice(roon_server, player_data)
            media_players.add(dev_id)
            entities = [
                media_player,
                RoonEventEntity(roon_server, player_data["display_name"]),
            ]
            async_add_entities(entities)
        else:
            # update existing player
            async_dispatcher_send(
                hass, f"room_media_player_update_{dev_id}", player_data
            )
```

If you feel strongly that event entities should only be created inside the event platform - then I'd need to create another callback inside `async_setup_entry` in the event platform.

Not sure this would be clearer - but happy to do so if you think it's better.

Reply:
Yes. Platforms aren't allowed to depend on another platform.